qiskit_optimization/converters/maximize_to_minimize.py

```diff
@@ -0,0 +1,78 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Converter to convert a maximization problem to minimization problem."""
+
+import copy
+import logging
+from typing import Optional, Union, Tuple, List
+
+import numpy as np
+
+from .quadratic_program_converter import QuadraticProgramConverter
+from ..exceptions import QiskitOptimizationError
+from ..problems.quadratic_objective import QuadraticObjective
+from ..problems.quadratic_program import QuadraticProgram
+
+logger = logging.getLogger(__name__)
+
+
+class MaximizeToMinimize(QuadraticProgramConverter):
+    """Convert a maximization problem to minimization problem."""
+
+    def __init__(self) -> None:
+        self._src = None  # type: Optional[QuadraticProgram]
+        self._dst = None  # type: Optional[QuadraticProgram]
+
+    def convert(self, problem: QuadraticProgram) -> QuadraticProgram:
+        """Convert a problem into a new minimization problem.
+
+        Args:
+            problem: The problem to be solved, that is a maximization or minimization problem.
+
+        Returns:
+            The converted problem, that is a minimization problem.
+        """
+
+        # Copy original QP as reference.
+        self._src = copy.deepcopy(problem)
+        self._dst = copy.deepcopy(problem)
```

Comment:
Is there really a need to store these? In terms of _dst its just returned from convert here. And for _src, the only place I see it used later is in interpret, where instead we could potentially simply just store problem.get_num_vars() rather than a copy of the whole original src QP.

Reply:
I was also thinking whether this is needed at all but have chosen an implementation which is consistent with the existing converters. All of them store self._src and self._dst. 
LinearEqualityToPenalty saves self._dst and it is not used later(as far as I know). The same goes for IntegerToBinary: even if no integer variables exist the problem is copied. I thought this is done to avoid side effects or something I don`t see.

If this is not the case, I would just store the number of variables as proposed.

Reply:
Ok, consistency is good. Maybe @t-imamichi has input here - it just seems to me a bit inefficient storing things like that when not really needed.

Reply:
Yes, some converters may store problems inefficiently. I would like to improve them in the future. @tomtuamnuq can avoid inefficient store here. For example, storing only get_num_vars or a shallow copy `self._src = problem`.

Reply:
I`ll only store num_vars. Should converters always return a new instance @t-imamichi in ``convert`` ?  Or can I just return the original problem if it is minimization? That would avoid copying.

Reply:
If minimization, you can return the original.

Reply:
MaximizeToMinimize only stores the number of variables now and avoids making a copy if the original problem is already a minimization problem.